Nimble/Matchers/AllPass.swift

```diff
@@ -0,0 +1,84 @@
+        return allPass("pass a condition", passFunc)
+}
+
+public func allPass<T,U where U: SequenceType, U.Generator.Element == T>
+    (passName:String, passFunc: (T?) -> Bool) -> NonNilMatcherFunc<U> {
+        return createAllPassMatcher() {
+            expression, failureMessage in
+            failureMessage.postfixMessage = passName
+            return passFunc(expression.evaluate())
+        }
+}
+
+public func allPass<T,U where U: SequenceType, U.Generator.Element == T>
+    (matcher: NonNilMatcherFunc<T>) -> NonNilMatcherFunc<U> {
+        return createAllPassMatcher() {matcher.matches($0, failureMessage: $1)}
+}
+
+private func createAllPassMatcher<T,U where U: SequenceType, U.Generator.Element == T>
+    (elementEvaluator:(Expression<T>, FailureMessage) -> Bool) -> NonNilMatcherFunc<U> {
+        return NonNilMatcherFunc { actualExpression, failureMessage in
+            failureMessage.actualValue = nil
+            if let actualValue = actualExpression.evaluate() {
+                for currentElement in actualValue {
+                    let exp = Expression(
+                        expression: {currentElement}, location: actualExpression.location)
+                    if !elementEvaluator(exp, failureMessage) {
+                        failureMessage.postfixMessage =
+                            "all \(failureMessage.postfixMessage),"
+                            + " but failed first at element <\(stringify(currentElement))>"
+                            + " in <\(stringify(actualValue))>"
+                        return false
+                    }
+                }
+            }
+            failureMessage.postfixMessage = "all \(failureMessage.postfixMessage)"
+            
+            return true
+        }
+}
+
+extension NMBObjCMatcher {
+    public class func allPassMatcher(matcher: NMBObjCMatcher) -> NMBObjCMatcher {
+        return NMBObjCMatcher(canMatchNil: false) { actualExpression, failureMessage, location in
+            let actualValue = actualExpression.evaluate()
+            var nsObjects = [NSObject]()
+            
+            var collectionIsUsable = true
+            if let value = actualValue as? NSFastEnumeration {
+                let generator = NSFastGenerator(value)
+                while let obj:AnyObject = generator.next() {
+                    if let nsObject = obj as? NSObject {
+                        nsObjects.append(nsObject)
+                    } else {
+                        collectionIsUsable = false
+                        break
+                    }
+                }
+            } else {
+                collectionIsUsable = false
+            }
+            
+            if !collectionIsUsable {
+                failureMessage.postfixMessage =
+                  "allPass only works with NSFastEnumeration (NSArray, NSSet, ...) of NSObjects"
+                failureMessage.expected = ""
+                failureMessage.to = ""
+                return false
+            }
```

Comment:
Unfortunately there are two situations at which the collection becomes unusable:
  * it contains at least one object that isn't a NSObject (line 58)
  * the collection isn't a NSFastEnumeration

Inlining would double the same code.
It could be moved into a closure that would be called at both places, but that would add a variable for the closure and would make the whole thing more confusing.